tracer/src/Datadog.Trace/Activity/ActivityListenerHandler.cs

```diff
@@ -48,16 +48,34 @@ public static bool OnShouldListenTo<T>(T source)
             {
                 if (source is IActivitySource activitySource)
                 {
-                    if (handler.ShouldListenTo(sName, activitySource.Version) && HandlerBySource.TryAdd(sName, handler))
+                    if (handler.ShouldListenTo(sName, activitySource.Version))
                     {
-                        Log.Debug("ActivityListenerHandler: {SourceName} will be handled by {Handler}.", sName, handler);
-                        return true;
+                        if (handler is DisableActivityHandler)
```

Comment:
Rather than checking the type of the handler here, I wonder if it makes more sense to change the return from `ShouldListenTo` to be an enum instead e.g. something like this (I hate all of these names though!)

```csharp
public enum ListenState
{
    Listen,
    Skip, // don't listen, but allow something else to listen
    Deny, // don't listen, don't allow anything else to listen
}
```